watsonsdk/AuthConfiguration.h

```diff
@@ -20,12 +20,14 @@
 
 @property NSString* basicAuthUsername;
 @property NSString* basicAuthPassword;
+@property (nonatomic) BOOL xWatsonLearningOptOut;
 
 @property (readonly) NSString *token;
 @property (copy, nonatomic) void (^tokenGenerator) (void (^tokenHandler)(NSString *token));
 
 - (void) invalidateToken;
 - (void) requestToken: (void(^)(AuthConfiguration *config)) completionHandler;
 - (NSDictionary*) createRequestHeaders;
+- (NSDictionary*) createRequestHeadersWithXWatsonLearningOptOut;
```

Comment:
I see. What do you think about having a private header "AuthConfiguration+Internal.h" ? It should look something like this:

```
// AuthConfiguration+Internal.h
#import AuthConfiguration.h
@interface AuthConfiguration(Internal)
- (NSDictionary*) createRequestHeaders;
- (NSDictionary*) createRequestHeadersWithXWatsonLearningOptOut;
@end
```
This way we can separate internal and internal stuff. In this case `createRequestHeadersWithXWatsonLearningOptOut` is used in STT and TTS request. IMO is better to have it as a method instead of doing it separately twice.